### PR TITLE
Remote Edge data store docs

### PR DESCRIPTION
**Pull request by jakub-dzon**
Signed-off-by: Jakub Dzon <jdzon@redhat.com>

### CODE REVIEW

File: documentation/latest/operations/remote_data_store.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+---
+layout: documentation
+---
+
+# Remote data store for Flotta Edge API
+
+## Introduction
+
+By default Flotta Edge API uses Kubernetes and CRD-based storage as a source of truth about managed Edge entities -
+`EdgeDevice` and `EdgeWorkload`. All the other CRDs serve as support for better device fleet management in Kubernetes
```
Comment (masayag) [inline documentation/latest/operations/remote_data_store.md:10]: s/fleet management/management (to avoid confusion with saas)

Comment (jakub-dzon) [reply documentation/latest/operations/remote_data_store.md]: Done.

File: documentation/latest/operations/remote_data_store.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  and
+  a [fake server implementation](https://github.com/project-flotta/flotta-operator/blob/main/internal/edgeapi/backend/remote/fake-server_test.go)
+  ;
+* Flotta Edge API deployed - as part of the whole Flotta deployment or a standalone deployment (Flotta operator would be
+  useless);
+* Edge Configuration API server has to be reachable from Flotta Edge API pod;
+
+### Flotta Edge API configuration
+
+Flotta Edge API configuration can be found in `flotta-edge-api-config` ConfigMap in `flotta` namespace. To configure
```
Comment (masayag) [inline documentation/latest/operations/remote_data_store.md:49]: is there a planned PR for that? I haven't found a reference to this configmap in our operator's repository or in code.

File: documentation/latest/operations/remote_data_store.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+## Introduction
+
+By default Flotta Edge API uses Kubernetes and CRD-based storage as a source of truth about managed Edge entities -
+`EdgeDevice` and `EdgeWorkload`. All the other CRDs serve as support for better device management in Kubernetes
+environment.
+
+![](../diagrams/edge_api.png)
+
+Users might want to store their device configuration data in some other place than Kubernetes etcd. To support that
+use-case we provide a way to use that external data store (or to be precise - data provider service) in Flotta Edge API.
```
[the remark this answers is not in the document]
Comment (jakub-dzon) [reply documentation/latest/operations/remote_data_store.md:16]: I don't think it is needed to split that interface - it is all used in one place by one caller.
I will, of course, update docs to be more precise.

File: documentation/latest/operations/remote_data_store.md (hunk 2 above)
Comment (jakub-dzon) [reply documentation/latest/operations/remote_data_store.md:49]: It's there: https://github.com/project-flotta/flotta-operator/blob/3ea8160dcadab2a556c9e179bd5ddc8bc3a3d458/config/edge-api/kustomization.yaml#L7

File: documentation/latest/operations/remote_data_store.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (jakub-dzon) [reply documentation/latest/operations/remote_data_store.md]: Done. WDYT now?

File: documentation/latest/operations/remote_data_store.md (hunk 2 above)
Comment (masayag) [reply documentation/latest/operations/remote_data_store.md:53]: right. was looking for the name post-processing by kustomize :-)

**Review by masayag — APPROVED** [review]

File: documentation/latest/operations/remote_data_store.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (masayag) [reply documentation/latest/operations/remote_data_store.md]: LGTM. Thank you!

**Comment by jakub-dzon** [comment]
> I need to update config options page as well.

Done.

File: documentation/latest/operations/deployment_options.md [hunk 4]
```diff
@@ -79,3 +79,81 @@ no need to have human interaction to start running workloads on any device.
...
+
+Edge API has different deployment modes, where some parameters can be turned
+on/off based on a configMap value. To edit one of these parameters, the
+following configmap should be edited.
+
+```shell
+kubectl edit cm -n flotta flotta-edge-api-config
+```
+
+After any change in that file, flotta-operator and API need to be rebooted.
```
Comment (masayag) [inline documentation/latest/operations/deployment_options.md:93]: Why a reboot is needed for flotta-operator?

Comment (jakub-dzon) [reply documentation/latest/operations/deployment_options.md:93]: :facepalm: copy & paste. On it.

Comment (jakub-dzon) [reply documentation/latest/operations/deployment_options.md]: Fixed.

**Review by pkliczewski — COMMENTED** [review]
/lgtm

**Pull request merged**